extensions/amp-story/1.0/amp-story-quiz.js

```diff
@@ -299,13 +352,130 @@ export class AmpStoryQuiz extends AMP.BaseElement {
    * @private
    */
   handleOptionSelection_(optionEl) {
-    this.triggerAnalytics_(optionEl);
+    this.responseDataPromise_.then(() => {
+      this.triggerAnalytics_(optionEl);
 
-    this.mutateElement(() => {
-      optionEl.classList.add('i-amphtml-story-quiz-option-selected');
-      this.quizEl_.classList.add('i-amphtml-story-quiz-post-selection');
+      this.mutateElement(() => {
+        optionEl.classList.add('i-amphtml-story-quiz-option-selected');
+        this.quizEl_.classList.add('i-amphtml-story-quiz-post-selection');
+
+        this.hasUserSelection_ = true;
+      });
+
+      this.updateReactionData_(optionEl.optionIndex_);
+    });
+  }
+
+  /**
+   * Get the Reaction data from the datastore
+   *
+   * @return {Promise<JsonObject>}
+   * @private
+   */
+  retrieveReactionData_() {
+    return this.executeReactionRequest_()
+      .then(response => this.handleSuccessfulDataRetrieval_(response))
+      .catch(error => {
+        if (error === ENDPOINT_UNAVAILABLE_ERROR) {
+          return;
+        }
+        dev().error(TAG, error);
+      });
+  }
 
-      this.hasReceivedResponse_ = true;
+  /**
+   * Update the Reaction data in the datastore
+   *
+   * @param {number} reactionValue
+   * @private
+   */
+  updateReactionData_(reactionValue) {
+    this.executeReactionRequest_(reactionValue).catch(error => {
+      if (error === ENDPOINT_UNAVAILABLE_ERROR) {
+        return;
```

Comment:
should we add a `user().error(TAG, error)` here to let publishers know what the error is?

Reply:
I left this out because otherwise all quizzes without endpoints would be throwing errors, but that shouldn't be an error case.

That leads me to think that I should instead be resolving the promise if there's no endpoint, sending different data through, and evaluating the case in a callback instead of handling it as an ignored error.

Changed to doing this way